Block/Js.php

```diff
@@ -102,7 +102,11 @@ public function getConnectJsUrl()
      */
     public function getCheckoutKey()
     {
-        return $this->configHelper->getAnyPublishableKey();
+        if($this->_request->getFullActionName() == Config::CHECKOUT_PAGE_ACTION){
+            return $this->configHelper->getPublishableKeyPayment();
```

Comment:
@johnmccombs1 there's no payment-only division for majority of the merchants. So no publishable key as well. This would affect Bolt logic on the checkout page, setting up track.js and connect.js, configuring the checkout button in minicart if available. When the plugin was initially developed there was an option to dynamically change the publishable key on the page depending on the frontend route - shipping or payment. On the shipping step the multi-step was used, and payment only on the payment form. This functionality was removed on Bolt side so we can not support it anymore. I guess that is one of the reasons we do not support magento default (payment only) checkout.
Perhaps we should discuss this on a call. cc @matt-thomason @YevhenShBolt @njfaries @ethanwayda22

Reply:
@danac-gs, Nature Hills Nursery wants to use Bolt in both the cart and the checkout page. See this task for more information about the request https://app.asana.com/0/564264490825835/1140792639357263

On Magento 1, we can both have divisions working on the cart and the checkout page and I used both of the divisions for several M1 merchants in the past: Akira and  Shop Rescue Spa.

Can you explain to me why this PR causes the issues on the checkout page on M2? The mini cart shouldn’t be used in the checkout page, right?

Reply:
@johnmccombs1 
1. I agree the minicart should be removed from the checkout page. However, by default it is shown and so it can be for some live stores.
Proposal: merchants that want to use Bolt on the default checkout should have minicart disabled on the page.
2. My concern was more regarding backward compatibility and loading track.js (we may or may not need connect.js depending on minicart, see #1 above) 
Proposal: I've just realized that we have `Enable Payment Only Checkout` configuration option, false by default. You can include it in the condition, alongside 
`$this->_request->getFullActionName() == Config::CHECKOUT_PAGE_ACTION`
and everything should be fine. This way the multi-step key would load as a fallback, as it does for the old stores, and track.js would work as expected.